src/client/public/scripts/components/base-element.ts

```diff
@@ -0,0 +1,18 @@
+import {render} from '../../../../../node_modules/lit-html/lib/lit-extended.js';
+import {TemplateResult} from '../../../../../node_modules/lit-html/lit-html.js';
+
+export abstract class BaseElement extends HTMLElement {
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    this.renderCallback();
+  }
+
+  renderCallback() {
```

Comment:
You may want to consider debouncing invocations of `renderCallback` to avoid multiple re-renders within the same turn.

Reply:
Yep, what's a good way of doing this? I saw you wrote your own microtask thing it seems?